libsolidity/formal/BMC.cpp

```diff
@@ -674,15 +696,21 @@ void BMC::visitAddMulMod(FunctionCall const& _funCall)
 	SMTEncoder::visitAddMulMod(_funCall);
 }
 
-void BMC::inlineFunctionCall(FunctionCall const& _funCall)
+void BMC::inlineFunctionCall(
+	FunctionDefinition const* _funDef,
+	Expression const& _callStackExpr,
+	Expression const* _calledExpr,
```

Comment:
It looks to me like we are not consistent about what `_calledExpr` represents.
In case of `Operations`, you pass the pointer to the whole operation expression, e.g. `a + b`.
But in case of a function call, e.g., `f(a,b)`, this is only the pointer to the expression representing the name of the function, i.e, `f`.

Reply:
Yea there is a mix up here:

- `pushCallStack` and `defineExpr` inside `createReturnedExpression` should be called on the `FunctionCall` object for the normal function call case, and indeed for the expression in case if user-defined operator call (e.g. `a + b`).
- `_funType->hasBoundFirstArgument()` can only be true for normal function calls (I assume, we should check in solidity-dev), so in `symbolicArguments` we do need the `FunctionCall::expression()` object (of type `Expression`) for the function call case, but not for the user-defined case

Reply:
Please correct me if I'm wrong, but I think now the arguments provided to `BMC::inlineFuncitonCall` are correct.